src/ranch_ssl.erl

```diff
@@ -298,24 +310,22 @@ cleanup(#{socket_opts:=SocketOpts}) ->
 cleanup(_) ->
 	ok.
 
--spec strip_unsupported_options(opts()) -> opts().
-strip_unsupported_options(SocketOpts) ->
-    Versions1 = lists:keyfind(versions, 1, SocketOpts),
-    Versions2 = lists:keyfind(protocol_versions, 1, SocketOpts),
-    if
-        (Versions1 == {versions, ['tlsv1.3']}) or (Versions2 == {protocol_versions, ['tlsv1.3']}) ->
-            NewSocketOpts = lists:filter(fun({X, _}) ->
-                    (X /= secure_renegotiate) and (X /= reuse_sessions) and (X /= next_protocols_advertised) and (X /= alpn_preferred_protocols);
-                (_) ->
-                    true
-                end, SocketOpts),
-            if
-                NewSocketOpts /= SocketOpts ->
-                    error_logger:warning_msg("~p~n dropping options unsupported by TLS1.3-only ssl sockets: " ++
-                    "secure_renegotiate, reuse_sessions, next_protocols_advertised and/or alpn_preferred_protocols from ~p~n", [?MODULE, SocketOpts])
-            end,
-            NewSocketOpts;
-        true ->
-            SocketOpts
-    end.
+get_tls_versions(SocketOpts) ->
+	case lists:keyfind(versions, 1, SocketOpts) of
+		{versions, Versions} ->
+			Versions;
+		false ->
+			get_tls_versions_env()
+	end.
+
+get_tls_versions_env() ->
+	case application:get_env(ssl, protocol_versions) of
```

Comment:
I think is is called `protocol_version` in the env.

Reply:
Indeed, fixed :)